src/main/scala/performanceanalysis/logreceiver/LogReceiverActor.scala

```diff
@@ -3,41 +3,47 @@ package performanceanalysis.logreceiver
 import akka.actor.{Actor, ActorLogging, ActorRef, Props}
 import performanceanalysis.server.Protocol._
 
+import scala.collection.mutable
+
 /**
   * Created by Jordi on 5-4-2016.
   */
 
 object LogReceiverActor {
-
   def props: Props = Props(new LogReceiverActor)
+
+  /**
+    * This function returns an array of lines. It treats
+    * \r\n, \n, \r as a new line
+    * @param input to be split
+    * @return array of lines
+    */
+  def splitIntoLines(input: String):List[String] = {
+    input.split("[\r\n]+").toList
+  }
 }
 
 class LogReceiverActor extends Actor with ActorLogging {
   private type ComponentId = String
   private type LogParser = ActorRef
+  private val logParsersById: mutable.Map[ComponentId, LogParser] = mutable.Map()
 
-  def receive: Receive = normal(Map())
-
-  def normal(logParsersById: Map[ComponentId, LogParser]): Receive = {
-    case msg: SubmitLog =>
-      handleSubmitLog(logParsersById, msg)
+  def receive: Receive = {
+    case SubmitLogs(componentId, logLines) =>
+      log.debug("Logs submitted with {}", componentId)
+      handleSubmitLog(componentId, logLines)
     case RegisterNewLogParser(compId, newLogParser) =>
-      log.info("New LogParser created with {}", compId)
-      handleRegisterNewLogParser(logParsersById, compId, newLogParser)
+      log.debug("New LogParser created with {}", compId)
+      logParsersById += compId -> newLogParser
   }
 
-  private def handleSubmitLog(logParsersById: Map[ComponentId, LogParser], msg: SubmitLog) = {
-    logParsersById.get(msg.componentId) match {
-      case None => sender() ! LogParserNotFound(msg.componentId)
+  private def handleSubmitLog(compId: ComponentId, logLines: String) = {
+    import LogReceiverActor.splitIntoLines
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Put imports in the beginning of the package](https://www.codacy.com/app/scala-academy/performance-analysis/file/2632818442/issues/source?bid=3220915&fileBranchId=3313889#l41)